DSharpPlus/Net/Rest/MultipartRestRequest.cs

```diff
@@ -103,8 +105,29 @@ public HttpRequestMessage Build()
             {
                 DiscordMessageFile current = this.Files[i];
 
-                StreamContent file = new(current.Stream);
+                ArrayPoolBufferWriter<byte> writer;
 
+                try
+                {
+                    writer = new ArrayPoolBufferWriter<byte>((int)current.Stream.Length);
+                }
+                catch

```

Comment:
according to further discussion, only catch NotSupportedException here: https://canary.discord.com/channels/379378609942560770/379386901725052928/1264010352409182369